catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/history/HistorianTest.java

```diff
@@ -0,0 +1,462 @@
+
+        assertThat(update, equalTo(metacard));
+    }
+
+    @Test(expected = IngestException.class)
+    public void testTryCommitStorageException()
+            throws StorageException, UnsupportedQueryException, SourceUnavailableException,
+            IngestException, URISyntaxException {
+        List<Metacard> metacards = getMetacardUpdatePair();
+
+        // Mock out a bad storage provider
+        StorageProvider exceptionStorageProvider = mock(StorageProvider.class);
+        doThrow(StorageException.class).when(exceptionStorageProvider)
+                .commit(any());
+
+        ContentItem item = mock(ContentItem.class);
+        when(item.getId()).thenReturn(METACARD_ID);
+        when(item.getUri()).thenReturn(RESOURCE_URI);
+        when(item.getMetacard()).thenReturn(metacards.get(0));
+
+        ReadStorageResponse readStorageResponse = mock(ReadStorageResponse.class);
+        when(readStorageResponse.getContentItem()).thenReturn(item);
+        when(exceptionStorageProvider.read(any())).thenReturn(readStorageResponse);
+
+        when(exceptionStorageProvider.create(any())).thenReturn(mock(CreateStorageResponse.class));
+
+        historian.setStorageProviders(Collections.singletonList(exceptionStorageProvider));
+
+        // Parameters for historian
+        UpdateStorageRequest storageRequest = mock(UpdateStorageRequest.class);
+        UpdateStorageResponse storageResponse = mock(UpdateStorageResponse.class);
+        UpdateResponse updateResponse = mock(UpdateResponse.class);
+
+        // send a request to update the metacard
+        updateMetacard(storageRequest, storageResponse, metacards.get(1));
+
+        mockQuery(metacards.get(1));
+        historian.version(storageRequest, storageResponse, updateResponse);
+    }
+
+    @Test
+    public void testBadContentItemSize()
+            throws StorageException, UnsupportedQueryException, SourceUnavailableException,
+            IngestException, URISyntaxException, IOException {
+        // The metacard and updated metacard
+        List<Metacard> metacards = getMetacardUpdatePair();
+
+        // Parameters for historian
+        UpdateStorageRequest storageRequest = mock(UpdateStorageRequest.class);
+        UpdateStorageResponse storageResponse = mock(UpdateStorageResponse.class);
+        UpdateResponse updateResponse = mock(UpdateResponse.class);
+
+        storeMetacard(metacards.get(0));
+
+        // send a request to update the metacard
+        updateMetacard(storageRequest, storageResponse, metacards.get(1));
+        storageProvider.update(storageRequest);
+        when(storageProvider.storageMap.get(RESOURCE_URI)
+                .getSize()).thenThrow(IOException.class);
+
+        mockQuery(metacards.get(1));
+        historian.version(storageRequest, storageResponse, updateResponse);
+
+        // Verify that the metacard updated
+        Metacard update = readMetacard();
+
+        assertThat(update, equalTo(metacards.get(1)));
+    }
+
```

Comment:
❓Do we have any assertions in the tests to verify that the resource URI are properly set on the metacards (set by `setResourceUriForContent()` method)?